### ISSUE TITLE
janitor bug in door

### ISSUE THREAD

**Issue by Waxime64** [CorsixTH/CorsixTH#657]
One of my jaynitor is blooked in door. I try to fire him, but nothing. The game always froze. I have a save with the bug if that help you

Created game with savegame version 91.
Savegame version is 91 (0.40), originally it was 91 (0.40)
Warning: Trying to remove nonexistant room build callback (function: 0x106e1c18) from humanoid (table: 0x106f0070).
Warning: Trying to remove nonexistant room remove callback (function: 0x106e1b68) from humanoid (table: 0x106f0070).
Savegame version is 91 (0.40), originally it was 91 (0.40)
Savegame version is 91 (0.40), originally it was 91 (0.40)
Error in timer handler: 
C:\Program Files\CorsixTH\Lua\entity.lua:130: Suspected infinite loop
stack traceback:
    C:\Program Files\CorsixTH\Lua\entity.lua: in function 'setTile'
    C:\Program Files\CorsixTH\Lua\entity.lua:171: in function 'setTilePositionSpeed'
    C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:175: in function 'timer_function'
    C:\Program Files\CorsixTH\Lua\entity.lua:215: in function 'tick'
    C:\Program Files\CorsixTH\Lua\entities\staff.lua:146: in function 'tick'
    C:\Program Files\CorsixTH\Lua\world.lua:1103: in function 'onTick'
    C:\Program Files\CorsixTH\Lua\app.lua:852: in function 'dispatch'
    C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>
Recovering from error in timer handler...
Warning: Trying to remove nonexistant room build callback (function: 0x16506a30) from humanoid (table: 0x0ffc9a18).
Warning: Trying to remove nonexistant room remove callback (function: 0x16503028) from humanoid (table: 0x0ffc9a18).
Savegame version is 91 (0.40), originally it was 91 (0.40)
Warning: Trying to remove nonexistant room build callback (function: 0x163b7358) from humanoid (table: 0x1661e420).
Warning: Trying to remove nonexistant room remove callback (function: 0x16818fd8) from humanoid (table: 0x1661e420).
Warning: Trying to remove nonexistant room build callback (function: 0x165c2918) from humanoid (table: 0x0f6bb640).
Warning: Trying to remove nonexistant room remove callback (function: 0x168a95f0) from humanoid (table: 0x0f6bb640).
Error in timer handler: 
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:395: Suspected infinite loop
stack traceback:
    C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua: in function 'timer_function'
    C:\Program Files\CorsixTH\Lua\entity.lua:215: in function 'tick'
    C:\Program Files\CorsixTH\Lua\entities\staff.lua:146: in function 'tick'
    C:\Program Files\CorsixTH\Lua\world.lua:1103: in function 'onTick'
    C:\Program Files\CorsixTH\Lua\app.lua:852: in function 'dispatch'
    C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>
Recovering from error in timer handler...
Error in timer handler: 
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:161: Suspected infinite loop
stack traceback:
    C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua: in function 'timer_function'
    C:\Program Files\CorsixTH\Lua\entity.lua:215: in function 'tick'
    C:\Program Files\CorsixTH\Lua\world.lua:1103: in function 'onTick'
    C:\Program Files\CorsixTH\Lua\app.lua:852: in function 'dispatch'
    C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>
Recovering from error in timer handler...
Savegame version is 91 (0.40), originally it was 91 (0.40)
Error in timer handler: 
C:\Program Files\CorsixTH\Lua\entity_map.lua:79: Suspected infinite loop
stack traceback:
    C:\Program Files\CorsixTH\Lua\entity_map.lua: in function 'remove_entity_from_table'
    C:\Program Files\CorsixTH\Lua\entity_map.lua:85: in function 'removeEntity'
    C:\Program Files\CorsixTH\Lua\entity.lua:116: in function 'setTile'
    C:\Program Files\CorsixTH\Lua\entity.lua:171: in function 'setTilePositionSpeed'
    C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:175: in function 'timer_function'
    C:\Program Files\CorsixTH\Lua\entity.lua:215: in function 'tick'
    C:\Program Files\CorsixTH\Lua\entities\staff.lua:146: in function 'tick'
    C:\Program Files\CorsixTH\Lua\world.lua:1103: in function 'onTick'
    C:\Program Files\CorsixTH\Lua\app.lua:852: in function 'dispatch'
    C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>
Recovering from error in timer handler...
Error in buttonup handler: 
C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua:60: attempt to call local 'timer_function' (a nil value)
stack traceback:
    C:\Program Files\CorsixTH\Lua\humanoid_actions\walk.lua: in function 'on_interrupt'
    C:\Program Files\CorsixTH\Lua\entities\humanoid.lua:635: in function 'setNextAction'
    C:\Program Files\CorsixTH\Lua\dialogs\staff_dialog.lua:313: in function 'callback'
    C:\Program Files\CorsixTH\Lua\window.lua:645: in function 'handleClick'
    C:\Program Files\CorsixTH\Lua\window.lua:1406: in function 'onMouseUp'
    C:\Program Files\CorsixTH\Lua\dialogs\staff_dialog.lua:269: in function 'onMouseUp'
    C:\Program Files\CorsixTH\Lua\window.lua:1390: in function 'onMouseUp'
    C:\Program Files\CorsixTH\Lua\ui.lua:735: in function 'dispatch'
    C:\Program Files\CorsixTH\Lua\app.lua:753: in function <C:\Program Files\CorsixTH\Lua\app.lua:748>

**Comment by wolfy1339** [comment]
Yes, please post the save. It will help the developers recreate the situation and find out how it's caused.

**Comment by Waxime64** [comment]
https://mega.co.nz/#!jNdh2aSB!4m7PaAWwAYHDKZwh2qsLva09x775mAU_cmdqH5-pWf0

**Comment by MarkL1961** [comment]
Is this a duplicate of issue #563 ?

**Comment by TheCycoONE** [comment]
I don't think so, both have infinite loops but the stack trace is pretty different.

**Comment by wolfy1339** [comment]
This doesn't seem to be the same (maybe I'm wrong). Please post the savegame right before the issue happens

**Comment by TheCycoONE** [comment]
No his issue is unrelated. I believe there was another issue for his but I haven't been on a computer to look it up. Mugmuggy was working on that stuff

**Comment by mugmuggy** [comment]
> or the room cleaned up the pukes while he was sweeping it?

Probably the same issue in #1329/#1344, having removed the litter through some other process before the sweep action finished.

**Comment by lewri** [comment]
Does this still occur? The gamelog is very noisy here to know what exact error we're trying to look at.

**Comment by ARGAMX** [comment]
> Does this still occur? The gamelog is very noisy here to know what exact error we're trying to look at.

I never saw this bug since I started to play CorsixTH 0.65.0 (2021 release).
I propose to close this as resolved.
